src/lib/phdi-building-blocks/phdi_building_blocks/linkage.py

```diff
@@ -1,68 +1,100 @@
-            # Combine given and family name
-            recent_name = next(
-                (name for name in patient["name"] if name.get("use") == "official"),
-                patient["name"][0],
-            )
+        # Combine given and family name
+        recent_name = get_field_with_use(patient, "name", "official", 0)
+        name_parts = recent_name.get("given", []) + [recent_name.get("family")]
+        name_str = "-".join([n for n in name_parts if n])
 
-            name_parts = recent_name.get("given", []) + [recent_name.get("family")]
-            name_str = "-".join([n for n in name_parts if n])
+        # Compile one-line address string
+        address_line = ""
+        if "address" in patient:
+            address = get_field_with_use(patient, "address", "home", 0)
+            address_line = get_one_line_address(address)
 
-            # Compile one-line address string
-            address_line = ""
-            if "address" in patient:
-                address = next(
-                    (addr for addr in patient["address"] if addr.get("use") == "home"),
-                    patient["address"][0],
-                )
-                address_line = " ".join(address.get("line", []))
-                address_line += f" {address.get('city')}, {address.get('state')}"
-                if "postalCode" in address and address["postalCode"]:
-                    address_line += f" {address['postalCode']}"
+        # Generate and store unique hash code
+        link_str = name_str + "-" + patient["birthDate"] + "-" + address_line
+        hashcode = generate_hash_str(link_str, salt_str)
 
-            # Generate and store unique hash code
-            link_str = name_str + "-" + patient["birthDate"] + "-" + address_line
-            hashcode = generate_hash_str(link_str, salt_str)
+        if "identifier" not in patient:
+            patient["identifier"] = []
 
-            if "identifier" not in patient:
-                patient["identifier"] = []
+        patient["identifier"].append(
+            {
+                "value": hashcode,
+                # Note: this system value corresponds to the FHIR specification
+                # for a globally used / generated ID or UUID--the standard here
+                # is to make the use "temporary" even if it's not
+                "system": "urn:ietf:rfc:3986",
+                "use": "temp",
+            }
+        )
+    return bundle
 
-            patient["identifier"].append(
-                {
-                    "value": hashcode,
-                    # Note: this system value corresponds to the FHIR specification
-                    # for a globally used / generated ID or UUID--the standard here
-                    # is to make the use "temporary" even if it's not
-                    "system": "urn:ietf:rfc:3986",
-                    "use": "temp",
-                }
-            )
+
+def get_field_with_use(patient: dict, field: str, use: str, default_field: int) -> str:
```

Comment:
This function name is not intuitive to me, but that could be because I'm not fully familiar with HL7. Will "use" be an easily recognized term by people working with public health data and FHIR? 

I'd also change the `patient` parameter to something like `resource` since this code can likely be used for other `resources`. `data` might even be a better name for it, with a description of what it represents in the docstring.

Reply:
Bryan, yeah "use" is actually a word with a specific FHIR-related meaning. I've tried to expound on this a little bit more in the documentation for the function, but I've also implemented the requested generalizing nomenclature changes.